iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/DanglingLineData.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.util;
+
+import org.apache.commons.math3.complex.Complex;
+import com.powsybl.iidm.network.DanglingLine;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+public class DanglingLineData {
+
+    private final String id;
+
+    private final double r;
+    private final double x;
+    private final double g1;
+    private final double g2;
+    private final double b1;
+    private final double b2;
+
+    private final double u1;
+    private final double theta1;
+    private final double p0;
+    private final double q0;
+
+    double boundaryBusU;
+    double boundaryBusTheta;
+
+    public DanglingLineData(DanglingLine danglingLine) {
+
+        id = danglingLine.getId();
+        r = danglingLine.getR();
+        x = danglingLine.getX();
+        g1 = danglingLine.getG();
+        b1 = danglingLine.getB();
+        g2 = 0.0;
+        b2 = 0.0;
+        p0 = danglingLine.getP0();
+        q0 = danglingLine.getQ0();
+
+        u1 = getV(danglingLine);
+        theta1 = getTheta(danglingLine);
+
+        boundaryBusU = Double.NaN;
+        boundaryBusTheta = Double.NaN;
+
+        Complex v1 = new Complex(u1 * Math.cos(theta1), u1 * Math.sin(theta1));
+
+        Complex vBoundaryBus = new Complex(Double.NaN, Double.NaN);
+        if (p0 == 0.0 && q0 == 0.0) {
+            LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(r, x, 1.0, 0.0, 1.0, 0.0, new Complex(g1, b1), new Complex(g2, b2));
+            vBoundaryBus = adm.y21.multiply(v1).negate().divide(adm.y22);
+        } else {
+
+            // Two buses Loadflow
+            double pBoundaryBus = -p0;
```

Comment:
It is here an approximation ? As ````r```` and ````x```` are not equal to zero ?

Reply:
No, the exact voltage at the boundary bus it is calculated. It is a two bus Loadflow and can be solved as it is a second degree equation.

Reply:
I am going to trust you for this second degree equation resolution, I have tried but I am a bit lost with ````sigmar```` and ````sigmai```` part. But I understand well the equation system, so I think I am going to trust you.

Reply:
Try this link [Two-buses](http://elequant.com/wp-content/uploads/Trias-2002-Two-bus-loadflow-exact-developments.pdf) Toni Trias the father of HELM explains how to solve the two buses problem exactly (second degree equation) and how to solve it applying the HELM methodology to this particular case.

Reply:
Thanks a lot and I have succeed with that help. I wonder about the support of this resolution in Powsybl. Mabye, could you add the link you gave me in the comment of the code or more comments ? We don't have our background but we will need to support that code in the future.